### PR TITLE
High-level TesterBlockchain network bootstrap method

### PR DESCRIPTION
**Pull request by KPrasch**
We have a growing need to run test or scripts within the context of an already deployed nucypher network.

##### Proposed Solution

Overrides `TesterBlockchain.connect`; Disregarding arguments, and redirecting to the default PyEVM configuration with an `InMemoryContractRegistry` and deployer options set on `BlockchainDeployerInterface`.

##### Usage
```
testerchain, accounts = TesterBlockchain.bootstrap_network()
```

Once the nucypher network is deployed, you can use the `Agents` API as usual:
```
token_agent = NucypherTokenAgent(blockchain=testerchain)
miner_agent = MinerAgent(blockchain=testerchain)
policy_agent = PolicyAgent(blockchain=testerchain)
```

Included in automated testing in `estimate_gas.py`

### CODE REVIEW

File: nucypher/utilities/sandbox/blockchain.py [hunk 1]
```diff
@@ -48,10 +58,12 @@ class TesterBlockchain(Blockchain):
     Blockchain subclass with additional test utility methods and options.
     """
 
+    _PROVIDER_URI = 'tester://pyevm'
     _instance = NO_BLOCKCHAIN_AVAILABLE
     _test_account_cache = list()
+    _default_test_accounts = 10
 
-    def __init__(self, test_accounts=None, poa=True, airdrop=True, *args, **kwargs):
+    def __init__(self, test_accounts=_default_test_accounts, poa=True, airdrop=True, *args, **kwargs):
```
Comment (jMyles) [inline nucypher/utilities/sandbox/blockchain.py]: This is an odd pattern - what made you decide to do this?

Comment (KPrasch) [reply nucypher/utilities/sandbox/blockchain.py]: To add the `10`?  Just because we always expect there to be => 10 accounts in tests.

Comment (jMyles) [reply nucypher/utilities/sandbox/blockchain.py]: I meant the pattern of resolution here - you have `_default_test_accounts`, a class variable, but which is only ever used at class evaluation time to set the default of `__init__`... why not just either make it `10` in `__init__` or have `__init__` grab the class variable?

File: nucypher/utilities/sandbox/blockchain.py [hunk 2]
```diff
@@ -143,3 +155,24 @@ def time_travel(self, hours: int=None, seconds: int=None, periods: int=None):
...
+        solidity_compiler = SolidityCompiler(test_contract_dir=CONTRACT_ROOT)
+        memory_registry = InMemoryEthereumContractRegistry()
+        interface = BlockchainDeployerInterface(provider_uri=cls._PROVIDER_URI, compiler=solidity_compiler, registry=memory_registry)
+        testerchain = TesterBlockchain(interface=interface, test_accounts=cls._default_test_accounts, airdrop=False)
+        return testerchain
+
+    @classmethod
+    def bootstrap_network(cls) -> Tuple['TesterBlockchain', List[str]]:
+
+        def __deploy_contracts(testerchain: TesterBlockchain) -> None:
```
Comment (jMyles) [inline nucypher/utilities/sandbox/blockchain.py]: Why separate this logic into a function?  Why not just put these three lines in the place where this function is currently called?

Comment (KPrasch) [reply nucypher/utilities/sandbox/blockchain.py]: Honestly - I think this was re-purposed code - Yeah this doesn't need to be enclosed.

:heavy_check_mark:

File: nucypher/utilities/sandbox/blockchain.py (hunk 1 above)
Comment (KPrasch) [reply nucypher/utilities/sandbox/blockchain.py]: Hmm.. I *think* I want this at the instance level ultimately.  :heavy_check_mark:

File: tests/metrics/estimate_gas.py [hunk 3]
```diff
@@ -165,25 +128,29 @@ def estimate_gas(analyzer: AnalyzeGas = None) -> None:
     #
     # Setup
     #
+
     if AnalyzeGas is None:
```
Comment (cygnusv) [inline tests/metrics/estimate_gas.py]: Perhaps you mean this?
```suggestion
    if analyzer is None:
```
(This GH suggestion may be buggy, it seems it removes a white-space even if the number of spaces is correct. I'm not sure if the patch is wrong, or only the rendering here)

File: .gitignore [hunk 4]
```diff
@@ -32,7 +32,4 @@ variables*.yml
 *ansible/*.retry
 *ansible/inventory/*
 .env
-examples/heartbeat_demo/*.json
-examples/heartbeat_demo/*.msgpack
-examples/heartbeat_demo/doctor-files/
-examples/heartbeat_demo/alicia-files/
```
Comment (cygnusv) [inline .gitignore]: Don't remove this!

Comment (KPrasch) [reply .gitignore]: :heavy_check_mark: (restored)

File: tests/metrics/estimate_gas.py (hunk 3 above)
Comment (KPrasch) [reply tests/metrics/estimate_gas.py]: Whoops! :heavy_check_mark:

**Pull request merged**